app/src/main/kotlin/com/intive/picover/images/view/ImagesScreen.kt

```diff
@@ -0,0 +1,73 @@
+package com.intive.picover.images.view
+
+import android.net.Uri
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import com.intive.picover.common.loader.PicoverLoader
+import com.intive.picover.images.viewmodel.ImagesViewModel
+import com.skydoves.landscapist.coil.CoilImage
+
+@Composable
+fun ImagesScreen(viewModel: ImagesViewModel) {
+	val state by viewModel.uiState.collectAsState()
+
+	when (state) {
+		is ImagesViewModel.LatestFirebaseStorageState.Success -> {
+			val uris = (state as ImagesViewModel.LatestFirebaseStorageState.Success).uris
```

Comment:
is it casting needed?

Reply:
![image](https://user-images.githubusercontent.com/33199010/232215775-363bff28-00bf-4a80-af3c-93ab6e26afb3.png)

Reply:
if you assign `state` to some val in when (like `when (val variable = state)`, then casting would be unnecessary, but now as you've added the imports, for me it looks not so bad and can be as it is, @marcinWisnia as you've reacted to my comment, is it okay for you?